### PR TITLE
Erc721 support

### PR DESCRIPTION
**Pull request by bhargav55**

### CODE REVIEW

File: contracts/package.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
```
Comment (zjesko) [inline contracts/package.json]: do we need this file?

File: lib/forge-std
Comment (zjesko) [inline lib/forge-std:1]: lets try to maintain the same versions as the original codebase? what do you think?

File: contracts/validator-manager/ERC721/ERC721PoSValidatorManager.sol
Comment (zjesko) [inline contracts/validator-manager/ERC721/ERC721PoSValidatorManager.sol:1]: can we merge this and POSValidatorManager? similarly for ValidatorManager?

File: contracts/validator-manager/ERC721/ERC721TokenStakingManager.sol
Comment (zjesko) [inline contracts/validator-manager/ERC721/ERC721TokenStakingManager.sol:1]: place the file at contracts/validator-manager/ERC721TokenStakingManager.sol

File: contracts/validator-manager/ValidatorManager.sol [hunk 2]
```diff
@@ -542,6 +545,7 @@ abstract contract ValidatorManager is Initializable, ContextUpgradeable, IValida
         uint64 newValidatorWeight,
         uint64 oldValidatorWeight
     ) private {
+
```
Comment (zjesko) [inline contracts/validator-manager/ValidatorManager.sol:525]: nit: extra lines and similar at other places. This is repo seems to have solhint, did you try applying that maybe?

**Review by zjesko — CHANGES_REQUESTED** [review]

File: contracts/package.json (hunk 1 above)
Comment (bhargav55) [reply contracts/package.json]: ya no need

File: lib/forge-std
Comment (bhargav55) [reply lib/forge-std:1]: this is for console logs in contracts, will remove

File: contracts/validator-manager/ERC721/ERC721PoSValidatorManager.sol
Comment (bhargav55) [reply contracts/validator-manager/ERC721/ERC721PoSValidatorManager.sol:1]: there are different functionalities, especially we need to store tokenIDs for validators and delegators, so we need to separate this out, we can pull out those specific methods from POSValidatorManager to respective ERC20StakingManager and ERC721StakingManager but I dont want to change their contracts

File: contracts/validator-manager/ValidatorManager.sol (hunk 2 above)
Comment (bhargav55) [reply contracts/validator-manager/ValidatorManager.sol:525]: I didnt use linting, will check that

**Pull request merged**